test/server/cards/02_SHD/events/Evacuate.spec.ts

```diff
@@ -0,0 +1,68 @@
+describe('Evacuate', function() {
+    integration(function(contextRef) {
+        describe('Evacuate\'s ability', function() {
+            it('should return all units cards to hand', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['evacuate'],
+                        groundArena: ['pyke-sentinel'],
+                        spaceArena: ['cartel-spacer'],
+                        leader: { card: 'finn#this-is-a-rescue', deployed: false },
+                    },
+                    player2: {
+                        hand: ['death-star-stormtrooper'],
+                        groundArena: ['wampa'],
+                        spaceArena: ['imperial-interceptor'],
+                        leader: { card: 'grand-moff-tarkin#oversector-governor', deployed: false },
+                    }
+                });
+                const { context } = contextRef;
+
+                context.player1.clickCard('evacuate');
+                expect(context.player1.hand).toHaveSize(2);
+                expect(context.player1.hand).toContain(context.pykeSentinel);
+                expect(context.player1.hand).toContain(context.cartelSpacer);
+                expect(context.player1.hand).not.toContain(context.evacuate);
+                expect(context.player1.groundArena).toHaveSize(0);
+
+                expect(context.player2.hand).toHaveSize(3);
+                expect(context.player2.hand).toContain(context.deathStarStormtrooper);
+                expect(context.player2.hand).toContain(context.wampa);
+                expect(context.player2.hand).toContain(context.imperialInterceptor);
+                expect(context.player1.groundArena).toHaveSize(0);
+            });
+
+            it('should return all unit cards to hand and discard upgrades and ignore deployed leaders', function () {
```

Comment:
These tests are great! One request though, unfortunately we have a practical issue where the Jasmine tests are slow and heavy and we have limited Github Actions time in the free tier, so as a consequence we have to try and consolidate our tests as much as possible (though it's not great practice for sure). We plan to address it some time in the future.

For now though, I think these two test cases could be consolidated into one by just adding the upgrades and deployed leaders to the above case.

Reply:
consolidated the test, actually caught a bug too, I checked player1 ground arena account twice